relay-profiling/src/sample/v2.rs

```diff
@@ -95,6 +96,9 @@ impl ProfileData {
             return Err(ProfileError::NotEnoughSamples);
         }
 
+        // Remove NaN as a side-effect
+        self.sort_samples_by_timestamp();
```

Comment:
You can use `FiniteF64` for the timestamp instead which means you don't have `NaN`'s in the first place.